Wrappers/Python/ccpi/viewer/CILViewer.py

```diff
@@ -622,18 +633,41 @@ def startRenderLoop(self):
 
     def setInput3DData(self, imageData):
         self.img3D = imageData
+        
+        # Have to overwrite old volume and clipping planes if they
+        # were previously created:
         if self.volume_render_initialised:
-            if self.volume.GetVisibility():
-                if self.clipping_plane_initialised:
-                    self.planew.SetEnabled(False)
-                    self.volume.GetMapper().RemoveAllClippingPlanes()
-                    self.clipping_plane_initialised = False
-                self.installVolumeRenderActorPipeline()
-                self.volume.VisibilityOn()
-            else:
-                self.volume_render_initialised = False
+            if self.clipping_plane_initialised:
+                self.planew.SetEnabled(False)
+                self.volume.GetMapper().RemoveAllClippingPlanes()
```

Comment:
These 2 lines seem to be repeated below. Are they necessary?

Reply:
These first 2 lines seem to clear what is already there, then the next 2 seem to fix another issue with the camera position caused by the clipping plane.

Reply:
They aren't necessary, and have now been deleted.

Reply:
Creating the new clipping plane overrides the old one anyway so there is no point deleting them in these 2 lines.